### DISCUSSION TITLE
How to add multiple loader criteria to `with_loader_criteria()` function

### DISCUSSION THREAD

**Discussion by christianholland** [sqlalchemy/sqlalchemy#8990] [Usage Questions] [answered]
Hi there, 

I have stumpled in the documentation over the `with_loader_criteria()` function and find it quite useful for my work. However, in the [documentation](https://docs.sqlalchemy.org/en/20/orm/queryguide/api.html#sqlalchemy.orm.with_loader_criteria) there are only examples shown with single criteria, e.g. 

```
stmt = select(User).options(
    selectinload(User.addresses),
    with_loader_criteria(Address, where_criteria = Address.email_address != 'foo'))` <-- here
)
```

How can I add further criteria such as `Address.provider == 'gmail'`

Intuitively I have tried 

```
with_loader_criteria(Address, where_criteria = Address.email_address != 'foo' and Address.provider == 'gmail')
```
but this seems not to work eventhough the documentation says that the `where_criteria` argument expects a `bool`

Help would be highly appreciated,
Thanks

**Comment by CaselIT** [comment, marked as answer]
Hi,

You can use [and_](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.and_) or [or_](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.or_)

using `and` / `or` does not work since it will result in a boolean value, not in an expression that can be converted in sql

**Reply by christianholland** [reply]
Thanks @CaselIT, using `and_` solved my issue
